src/core/net/dnssd_server.cpp

```diff
@@ -1286,6 +1417,705 @@ void Server::ResetUpstreamQueryTransaction(UpstreamQueryTransaction &aTxn, Error
+
+    ConstructFullInstanceName(aResult.mServiceInstance, aResult.mServiceType, instanceName);
+    HandleResult(kResolvingSrv, instanceName, &Response::AppendSrvRecord, ProxyResult(aResult));
+
+exit:
+    return;
+}
+
+void Server::DiscoveryProxy::HandleTxtResult(otInstance *aInstance, const otPlatDnssdTxtResult *aResult)
+{
+    AsCoreType(aInstance).Get<Server>().mDiscoveryProxy.HandleTxtResult(*aResult);
+}
+
+void Server::DiscoveryProxy::HandleTxtResult(const Dnssd::TxtResult &aResult)
+{
+    Name::Buffer instanceName;
+
+    VerifyOrExit(mIsRunning);
+    VerifyOrExit(aResult.mTtl != 0);
+    VerifyOrExit(aResult.mInfraIfIndex == Get<BorderRouter::InfraIf>().GetIfIndex());
+
+    ConstructFullInstanceName(aResult.mServiceInstance, aResult.mServiceType, instanceName);
+    HandleResult(kResolvingTxt, instanceName, &Response::AppendTxtRecord, ProxyResult(aResult));
+
+exit:
+    return;
+}
+
+void Server::DiscoveryProxy::HandleIp6AddressResult(otInstance *aInstance, const otPlatDnssdAddressResult *aResult)
+{
+    AsCoreType(aInstance).Get<Server>().mDiscoveryProxy.HandleIp6AddressResult(*aResult);
+}
+
+void Server::DiscoveryProxy::HandleIp6AddressResult(const Dnssd::AddressResult &aResult)
+{
+    bool         hasValidAddress = false;
+    Name::Buffer fullHostName;
+
+    VerifyOrExit(mIsRunning);
+    VerifyOrExit(aResult.mInfraIfIndex == Get<BorderRouter::InfraIf>().GetIfIndex());
+
+    for (uint16_t index = 0; index < aResult.mAddressesLength; index++)
+    {
+        const Dnssd::AddressAndTtl &entry   = aResult.mAddresses[index];
+        const Ip6::Address         &address = AsCoreType(&entry.mAddress);
+
+        if (entry.mTtl == 0)
+        {
+            continue;
+        }
+
+        if (IsProxyAddressValid(address))
+        {
+            hasValidAddress = true;
+            break;
+        }
+    }
+
+    VerifyOrExit(hasValidAddress);
+
+    ConstructFullName(aResult.mHostName, fullHostName);
+    HandleResult(kResolvingIp6Address, fullHostName, &Response::AppendHostAddresses, ProxyResult(aResult));
+
+exit:
+    return;
+}
+
+void Server::DiscoveryProxy::HandleResult(ProxyAction         aAction,
+                                          const Name::Buffer &aName,
+                                          ResponseAppender    aAppender,
+                                          const ProxyResult  &aResult)
+{
+    // Common method that handles result from DNS-SD/mDNS. It
+    // iterates over all `ProxyQuery` entries and checks if any entry
+    // is waiting for the result of `aAction` for `aName`. Matching
+    // queries are updated using the `aAppender` method pointer,
+    // which appends the corresponding record(s) to the response. We
+    // then determine the next action to be performed for the
+    // `ProxyQuery` or if it can be finalized.
+
+    ProxyQueryList nextActionQueries;
+    ProxyQueryInfo info;
+    ProxyAction    nextAction;
```

Comment:
This has been reported on the previous PR: https://github.com/openthread/openthread/pull/9362#pullrequestreview-1934492901.

Reply:
That's a different variable! `action` vs `nextAction` :). The `action` one is indeed fixed in this branch.


The compiler is being too careful in case an `enum` value is not one of the specified value (which is technically possible and would be good to protect against but super unlikely). 
- Will fix in new push.
- Will also enable `DISCOVERY_PROXY` in the existing CI `scan-build` GH action (which will include similar checks)

Reply:
Fixed in new push.